src/server/bitops_family.cc

```diff
@@ -0,0 +1,267 @@
+std::string GetString(EngineShard* shard, const PrimeValue& pv);
+// ------------------------------------------------------------------------- //
+//  Bits manipulation functions
+constexpr int32_t GetBitIndex(int32_t offset) noexcept {
+  return offset % OFFSET_FACTOR;
+}
+
+constexpr int32_t GetByteIndex(int32_t offset) noexcept {
+  return offset / OFFSET_FACTOR;
+}
+
+std::string::value_type GetByteValue(const std::string& value, int32_t offset) {
+  return value[GetByteIndex(offset)];
+}
+
+long GetBitValue(const std::string& value, int32_t offset) {
+  const std::bitset<8> bits{static_cast<unsigned long long>(GetByteValue(value, offset))};
+  return static_cast<long>(bits.test(INDEX_MAPPING[GetBitIndex(offset)]));
+}
+
+long GetBitValueSafe(const std::string& value, int32_t offset) {
+  return ((value.size() * OFFSET_FACTOR) > (std::size_t)offset) ? GetBitValue(value, offset) : 0;
+}
+
+long SetBitValue(std::string& value, int32_t offset, int32_t new_value) {
+  // we need to return the old value after setting the value for offset
+  const auto old_value{GetBitValue(value, offset)};  // save this as the return value
+  std::bitset<8> bits{static_cast<unsigned long long>(GetByteValue(value, offset))};
+  bits.set(INDEX_MAPPING[GetBitIndex(offset)], new_value != 0);
+  value[GetByteIndex(offset)] = static_cast<std::string::value_type>(
+      bits.to_ulong());  // should be safe since this really holds only 8 bits
+  return old_value;
+}
+
+// ------------------------------------------------------------------------- //
+// Helper functions to access the data or change it
+
+class OverRideValue {
+  long old_value_ = -1;
+  SetCmd cmd_;
+  SetCmd::SetParams params_;
+
+ public:
+  OverRideValue(const OpArgs& args, DbIndex db_index, uint32_t memcache_flags)
+      : cmd_{args}, params_{db_index} {
+    params_.how =
+        SetCmd::SET_ALWAYS;  // we want to override what we had in case case, no matter what!
+    params_.memcache_flags = memcache_flags;
+  }
+
+  OpResult<long> Set(std::string_view key, int32_t offset, int32_t value);
+
+ private:
+  constexpr long OldValue() const {
+    return old_value_;
+  }
+
+  std::string GetValidNewValue(PrimeIterator it, int32_t offset, int32_t value) const;
+};
+
+std::string OverRideValue::GetValidNewValue(PrimeIterator it, int32_t offset, int32_t value) const {
+  if (IsValid(it)) {
+    const PrimeValue& pv = it->second;
+
+    std::string old_value{GetString(cmd_.args().shard, pv)};
+    if ((static_cast<int32_t>(old_value.size()) * OFFSET_FACTOR) <=
+        offset) {  // need to resize first
+      old_value.resize(GetByteIndex(offset) + 1, 0);
+    }
+    return old_value;
+  } else {  // simple case, just create new string with correct length and set it
+    std::string new_value(GetByteIndex(offset) + 1, 0);
+    return new_value;
+  }
+}
+
+OpResult<long> OverRideValue::Set(std::string_view key, int32_t offset, int32_t value) {
```

Comment:
the naming a little confusing. You have a function `GetValidNewValue` that returns string `entry`.
And you have a `value` which is int32_t and I guess you use as a bit. 
maybe call a bit value `bit` and entry rename to `strval` or blob or something ?

Reply:
ok

Reply:
changed to bit_value